mautrix/client/api/modules/media_repository.py

```diff
@@ -45,22 +48,15 @@ class MediaRepositoryMethods(BaseClientAPI):
     uploads of media.
     """
 
-    async def unstable_create_mxc(self) -> ContentURI:
+    async def unstable_create_mxc(self) -> MediaCreateResponse:
         """
         Create a media ID for uploading media to the homeserver. Requires the homeserver to have
         `MSC2246 <https://github.com/matrix-org/matrix-spec-proposals/pull/2246>`__ support.
 
         Returns:
-            The MXC URI that can be used to upload a file to later.
-
-        Raises:
-            MatrixResponseError: If the response does not contain a ``content_uri`` field.
+            MediaCreateResponse Containing the MXC URI that can be used to upload a file to later, as well as an optional upload URL
         """
-        resp = await self.api.request(Method.POST, MediaPath.unstable["fi.mau.msc2246"].create)
-        try:
-            return resp["content_uri"]
-        except KeyError:
-            raise MatrixResponseError("`content_uri` not in response.")
+        return await self.api.request(Method.POST, MediaPath.unstable["fi.mau.msc2246"].create)
```

Comment:
```suggestion
        return MediaCreateResponse.deserialize(await self.api.request(Method.POST, MediaPath.unstable["fi.mau.msc2246"].create))
```

(might also be nicer to have a temporary variable instead of a long line)